adapters/data_saver.py

```diff
@@ -32,3 +32,8 @@ def write(self, content: str, output_path: str) -> None:
             print(f'Scores saved to "{file_path}" successfully.')
         except OSError as e:
             raise FileSaveError(e)
+
+
+class TerminalPrinter(FileWriter):
+    def write(self, content: str, output_path: str = None) -> None:
```

Comment:
As you may have noticed, `output_path` is not needed by `TerminalPrinter`. This means, that the `output_path` is not part of an interface for writing text to selected output, but a feature of a particular writer. Therefore writer classes should have `__init__` function to store data specific for them (`output_path` for Pdf and TxtWriter, none for terminal) and the interface function `write` would be defined as......

Reply:
def write(self, content: str) -> None: